solutions/cc-AMONGUS2.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: cc-AMONGUS2
+source: CC
+title: Among Us
+author: Ryan Chou
+---
+
+## Explanation
+If we only have two players $x$ and $y$, then we have 4 cases possible cases. 
+
+* If $x$ vouches for $y$ and $x$ is an imposter, then $y$ must also be an imposter.
+
+* If $x$ vouches for $y$ and $x$ is a crewmate, then $y$ must also be a crewmate.
+
+* If $x$ accuses $y$ of being an imposter and $x$ is an imposter, then $y$ must be a crewmate.
+
+* If $x$ accuses $y$ of being an imposter and $x$ is a crewmate, then $y$ must be an imposter.
+
+You'll notice that whenever $x$ accuses $y$, they'll always be of a different type, and when $x$ vouches for $y$, they'll always be of the same type.
+
+With this information, we can build a bipartite graph and color nodes with the same color if they vouch for each other, or different colors if one of them accuses the other. If we can't build this graph, then no valid arrangement can exist.
+
+Otherwise, we'll consider each connected component independently and keep track of the number of crewmates and imposters. At the end, we'll return the sum of the maximum of these two values for all connected components. We don't care about the roles of each node because we can switch all of them to get another valid solution.
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <stack>
+using namespace std;
+
+int main() {
+	int t;
+	cin >> t;
+
+	for (int _ = 0; _ < t; _++) {
```

Comment:
why underscore instead of int i

Reply:
why i? it's a useless var anyway

Reply:
I usually use `tc` or something just to make it clear that you're iterating over the test cases.

Reply:
should be pretty obvious, but ok